packages/cli/src/workflow-execute-additional-data.ts

```diff
@@ -776,7 +776,41 @@ export async function executeWorkflow(
 	workflowInfo: IExecuteWorkflowInfo,
 	additionalData: IWorkflowExecuteAdditionalData,
 	options: ExecuteWorkflowOptions,
-): Promise<Array<INodeExecutionData[] | null> | IWorkflowExecuteProcess> {
+): Promise<ExecuteWorkflowData> {
+	const activeExecutions = Container.get(ActiveExecutions);
+
+	const workflowData =
+		options.loadedWorkflowData ??
+		(await getWorkflowData(workflowInfo, options.parentWorkflowId, options.parentWorkflowSettings));
+
+	const runData =
+		options.loadedRunData ??
+		(await getRunData(workflowData, options.inputData, options.startMetadata));
+
+	const executionId = await activeExecutions.add(runData);
+
+	const executionPromise = startExecution(
+		additionalData,
+		options,
+		executionId,
+		runData,
+		workflowData,
+	);
+
+	if (options.doNotWaitToFinish) {
+		return { executionId, data: [null] };
+	}
+
+	return await executionPromise;
```

Comment:
here to allow to skip waiting for subexecutions, for example in execute workflow now
<img width="447" alt="Screenshot 2024-11-12 at 13 00 00" src="https://github.com/user-attachments/assets/18b68808-2d47-447f-9931-19a7140c830a">